app/assets/javascripts/controllers/cloud_volume/cloud_volume_form_controller.js

```diff
@@ -67,6 +67,26 @@ ManageIQ.angular.app.controller('cloudVolumeFormController', ['$http', '$scope',
     miqService.miqAjaxButton(url);
   };
 
+  $scope.backupCreateClicked = function() {
+    var url = '/cloud_volume/backup_create/' + cloudVolumeFormId + '?button=create';
+    miqService.miqAjaxButton(url, $scope.cloudVolumeModel, { complete: false });
+  };
+
+  $scope.cancelBackupCreateClicked = function() {
+    var url = '/cloud_volume/backup_create/' + cloudVolumeFormId + '?button=cancel';
+    miqService.miqAjaxButton(url);
+  };
+
+  $scope.backupRestoreClicked = function() {
+    var url = '/cloud_volume/backup_restore/' + cloudVolumeFormId + '?button=restore';
+    miqService.miqAjaxButton(url, $scope.cloudVolumeModel, { complete: false });
+  };
+
+  $scope.cancelBackupRestoreClicked = function() {
+    var url = '/cloud_volume/backup_restore/' + cloudVolumeFormId + '?button=cancel';
+    miqService.miqAjaxButton(url);
+  };
+
```

Comment:
All these functions look so very similar. Can we deduplicate it? Higher order function maybe?